### PR TITLE
Add timeout parameter to workflow class

### PR DESCRIPTION
**Pull request by brent-johnson**
Enable workflow timeout parameters to be passed into workflow and task to set timeout warning notifications

Description
added timeout option to Workflow class

How Has This Been Tested?
tests updated to cover new functionality
dev test by deploying workflows to databricks with timeout_seconds

Screenshots (if appropriate):
Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: brickflow/engine/workflow.py [hunk 1]
```diff
@@ -117,8 +117,8 @@ class Workflow:
     schedule_pause_status: str = "UNPAUSED"
     default_cluster: Optional[Cluster] = None
     clusters: List[Cluster] = field(default_factory=lambda: [])
-
     health: Optional[List[JobsHealthRules]] = None
+    timeout_seconds: Optional[float] = None
```
Comment (stikkireddy) [inline brickflow/engine/workflow.py:121]: shouldnt this be int? and not a floating point number?

Comment (brent-johnson) [reply brickflow/engine/workflow.py:121]: @stikkireddy I defined it as a float based on your model: https://github.com/Nike-Inc/brickflow/blob/main/brickflow/bundles/model.py#L1664 would you like me to update the model as well?

Comment (brent-johnson) [reply brickflow/engine/workflow.py]: After a conversation with @asingamaneni I've changed this in the model from 'float' to 'int.' 
This did not effect unit tests passing even when values were left as floats.  

I want to test with a real actual deploy that has the value set as a float before this is merged.

Comment (brent-johnson) [reply brickflow/engine/workflow.py]: So I worked with Srinivas Caveri, we changed the Workflow and TaskSettings definition in the bundle model and in the Workflow class from 'float' to 'int'.

We then performed a deploy with a task timeout set to "2400.5" and the deploy went through and the task was set to 40 minutes and 0 seconds which was the expected outcome. 

That confirms that if any existing customer somehow had a timeout value set to a decimal number of seconds this change will be none disruptive to them.

File: tests/engine/test_workflow.py [hunk 2]
```diff
@@ -189,6 +189,9 @@ def test_health_settings(self):
             ]
         }
 
+    def test_timeout_seconds(self):
+        assert wf.timeout_seconds == 42.5
```
Comment (stikkireddy) [inline tests/engine/test_workflow.py:193]: probably should change this to an integer

Comment (brent-johnson) [reply tests/engine/test_workflow.py]: done

File: tests/engine/sample_workflow.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
             {"metric": "RUN_DURATION_SECONDS", "op": "GREATER_THAN", "value": 7200}
         ]
     },
+    timeout_seconds=42.5,
```
[the remark this answers is not in the document]
Comment (brent-johnson) [reply tests/engine/sample_workflow.py]: updated

**Review by stikkireddy — APPROVED** [review]
LGTM thanks for the contribution!

**Review by asingamaneni — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**